nav.go

```diff
@@ -221,23 +223,58 @@ func (dir *dir) sort() {
 	// of equivalent elements will be reversed
 	switch dir.sortby {
 	case naturalSort:
-		sort.SliceStable(dir.files, func(i, j int) bool {
+		fallback := func(i, j int) bool {
 			s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
 			if !dir.reverse {
 				return naturalLess(s1, s2)
 			} else {
 				return naturalLess(s2, s1)
 			}
-		})
+		}
+
+		if dir.locale != localeStrDisable {
+			if collator, err := makeCollator(dir.locale); err == nil {
+				sort.SliceStable(dir.files, func(i, j int) bool {
+					s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
+					if !dir.reverse {
+						return localeNaturalLess(collator, s1, s2)
+					} else {
+						return localeNaturalLess(collator, s2, s1)
+					}
+				})
+			} else {
+				sort.SliceStable(dir.files, fallback)
+			}
+		} else {
+			sort.SliceStable(dir.files, fallback)
+		}
```

Comment:
I'm not a fan of this fallback logic with multiple nested `if`/`else` blocks. Is it possible to instead use `language.Und` in the case where getting the language tag fails?

Reply:
Yes, I also think this nested condition is not appealing to eyes. Using `language.Und` is logical OK.

But note that, using `language.Und` means stating data being processed is not bind to any locale or is non-linguistic content, and will leads to `a` being smaller then `B`. If this is acceptable for application, then I'll be happy to make some change.

Reply:
Actually I realise now that `makeCollator` also checks for `localeStrDisable`, so you can change the structure of the code to this:

```go
if collator, err := makeCollator(dir.locale, collate.Numeric); err == nil {
    sort.SliceStable(dir.files, func(i, j int) bool {
        s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
        result := collator.CompareString(s1, s2)
        if !dir.reverse {
            return result < 0
        } else {
            return result > 0
        }
    })
} else {
    sort.SliceStable(dir.files, func(i, j int) bool {
        s1, s2 := normalize(dir.files[i].Name(), dir.files[j].Name(), dir.ignorecase, dir.ignoredia)
        if !dir.reverse {
            return naturalLess(s1, s2)
        } else {
            return naturalLess(s2, s1)
        }
    })
}
```